src/main/java/com/google/api/gax/grpc/ApiCallSettings.java

```diff
@@ -44,44 +44,70 @@
  */
 public class ApiCallSettings {
 
-  private Set<Status.Code> retryableCodes = new HashSet<>();
-  private RetryParams retryParams = null;
+  protected ImmutableSet<Status.Code> retryableCodes;
+  protected RetrySettings.Builder retrySettingsBuilder;
 
-  /**
-   * Sets the retryable codes.
-   */
-  public ApiCallSettings setRetryableCodes(Set<Status.Code> retryableCodes) {
-    this.retryableCodes = retryableCodes;
-    return this;
+  public ImmutableSet<Status.Code> getRetryableCodes() {
+    return retryableCodes;
   }
 
-  /**
-   * Sets the retryable codes.
-   */
-  public ApiCallSettings setRetryableCodes(Status.Code... codes) {
-    this.retryableCodes = Sets.newHashSet(codes);
-    return this;
+  public RetrySettings getRetrySettings() {
+    return retrySettingsBuilder.build();
   }
 
-  /**
-   * Gets the retryable codes.
-   */
-  public Set<Status.Code> getRetryableCodes() {
-    return retryableCodes;
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  public Builder toBuilder() {
+    return new Builder(this);
   }
 
-  /**
-   * Sets the retry params.
-   */
-  public ApiCallSettings setRetryParams(RetryParams retryParams) {
-    this.retryParams = retryParams;
-    return this;
+  protected ApiCallSettings(Builder builder) {
```

Comment:
The constructor shouldn't take a builder, it should take a parameter for each member.